### PR TITLE
raise IonQRetriableError when get fails

### PR DESCRIPTION
**Pull request by splch**
### Summary

try-catch block has been added around the network request to handle exceptions that can be retried

### CODE REVIEW

File: qiskit_ionq/ionq_client.py [hunk 1]
```diff
@@ -244,7 +246,12 @@ def get_results(
             params.update(extra_query_params)
 
         req_path = self.make_path("jobs", job_id, "results")
-        res = requests.get(req_path, params, headers=self.api_headers, timeout=30)
```
Comment (Cynocracy) [inline qiskit_ionq/ionq_client.py:247]: Can we also add to other requests.get calls in this file?

Ty!

**Review by Cynocracy — APPROVED** [review]

File: qiskit_ionq/ionq_client.py (hunk 1 above)
Comment (splch) [reply qiskit_ionq/ionq_client.py:247]: didnt mean to delete that haha nice catch

File: qiskit_ionq/ionq_client.py [hunk 2]
```diff
@@ -77,6 +77,32 @@ def make_path(self, *parts):
...
+            IonQRetriableError: When a retriable error occurs during the request.
+
+        Returns:
+            Response: A requests.Response object.
+        """
+        try:
+            res = requests.get(
+                req_path, params=params, headers=headers, timeout=timeout
+            )
+            res.raise_for_status()
```
Comment (Cynocracy) [inline qiskit_ionq/ionq_client.py:100]: You shouldn't call this here as the ionq specific raise_for_status is called later

File: qiskit_ionq/ionq_client.py [hunk 3]
```diff
@@ -77,6 +77,32 @@ def make_path(self, *parts):
...
+        except requests.exceptions.RequestException as req_exc:
```
Comment (Cynocracy) [inline qiskit_ionq/ionq_client.py:101]: if you remove raise_for_status you can just catch any error here, as they should all relate to the socket i/o

**Pull request merged**